src/v/model/panda_link.h

```diff
@@ -0,0 +1,134 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "serde/envelope.h"
+#include "serde/rw/variant.h"
+#include "utils/named_type.h"
+#include "utils/unresolved_address.h"
+#include "utils/uuid.h"
+
+#include <seastar/util/bool_class.hh>
+
+namespace model {
+/// ID of the panda link - used internally based off of controller offset
+using panda_link_id = named_type<int64_t, struct panda_link_id_tag>;
+/// UUID of the panda link - used externally
+using panda_link_uuid = named_type<uuid_t, struct panda_link_uuid_tag>;
+/// Name of the panda link
+using panda_link_name = named_type<ss::sstring, struct panda_link_name_tag>;
+
+/**
+ * @brief SCRAM credentials to use for authentication
+ */
+struct scram_credentials
+  : serde::
+      envelope<scram_credentials, serde::version<0>, serde::compat_version<0>> {
+    /// SCRAM username
+    ss::sstring username;
+    /// SCRAM password
+    ss::sstring password;
+    /// SASL-SCRAM mechanism to use
+    ss::sstring mechanism;
+
+    friend bool operator==(const scram_credentials&, const scram_credentials&)
+      = default;
+    auto serde_fields() { return std::tie(username, password, mechanism); }
+};
+
+/**
+ * @brief Represents the settings for connection to a remote cluster
+ */
+struct panda_link_connection
+  : serde::envelope<
+      panda_link_connection,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    /// List of addresses to bootstrap the connection
+    std::vector<net::unresolved_address> bootstrap_servers;
+    /// Support authn variants.  Currently only SCRAM but update this to add
+    /// support for OIDC or GSSAPI in the future.
+    using authn_variant = serde::variant<scram_credentials>;
+    /// Authentication configuration for the connection
+    std::optional<authn_variant> authn_config;
+    /// Path to certificate file to use
+    ss::sstring cert_file_path;
+    /// Path to key file (when mTLS is in use)
+    ss::sstring key_file_path;
+    /// Path to the CA file to use
+    ss::sstring ca_file_path;
+
+    friend bool
+    operator==(const panda_link_connection&, const panda_link_connection&)
+      = default;
+
+    auto serde_fields() {
+        return std::tie(
+          bootstrap_servers,
+          authn_config,
+          cert_file_path,
+          key_file_path,
+          ca_file_path);
+    }
+};
+
+struct panda_link_metadata
+  : serde::envelope<
+      panda_link_metadata,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    /// Name of the panda link
+    panda_link_name name;
+    /// Unique identifier for the panda link
+    panda_link_uuid uuid;
+    /// Connection settings for the panda link
+    panda_link_connection connection;
+    /// Type to indicate if the panda link is paused
+    using panda_link_paused_t = ss::bool_class<struct panda_link_paused_tag>;
```

Comment:
I think this can just be `paused_t` because it will be referenced as `panda_link_metadata::paused_t`